crates/storage/src/transactional.rs

```diff
@@ -656,6 +664,68 @@ mod test {
         use super::*;
         use crate::column::Column;
 
+        #[test]
+        fn read_returns_from_view_exact_size() {
+            // setup
+            let storage = InMemoryStorage::<Column>::default();
+            let mut view = storage.read_transaction();
+            let key = vec![0xA, 0xB, 0xC];
+            let expected = Value::from([1, 2, 3]);
+            view.put(&key, Column::Metadata, expected.clone()).unwrap();
+            // test
+            let mut buf = [0; 3];
+            let ret = view.read(&key, Column::Metadata, 0, &mut buf).unwrap();
+            // verify
+            assert_eq!(ret, Some(3));
+            assert_eq!(buf, [1, 2, 3]);
+        }
+
+        #[test]
+        fn read_returns_from_view_buf_smaller() {
+            // setup
+            let storage = InMemoryStorage::<Column>::default();
+            let mut view = storage.read_transaction();
+            let key = vec![0xA, 0xB, 0xC];
+            let expected = Value::from([1, 2, 3]);
+            view.put(&key, Column::Metadata, expected.clone()).unwrap();
+            // test
+            let mut buf = [0; 2];
+            let ret = view.read(&key, Column::Metadata, 0, &mut buf).unwrap();
+            // verify
+            assert_eq!(ret, Some(2));
+            assert_eq!(buf, [1, 2]);
+        }
```

Comment:
Maybe add another test with offset and passing. Could just be the same as this but 
`assert_eq!(buf, [2, 3]);`

Reply:
done